### PR TITLE
[got] Fix return type of json request

### PR DESCRIPTION
**Pull request by LinusU**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://github.com/sindresorhus/got/blob/8b040afc0749523fcaaf6db81702c27e74f7a1d2/index.js#L312
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

-----

When requesting JSON from remote servers, anything that is considered valid JSON by `JSON.parse` is returned as the `body` property. That includes JSON scalars (which I think at first was not valid as the root, but the spec was later amended by a different RFC to allow it...).

Since the return value of `JSON.parse` is typed as `any`, I opted for using the same here.

### CODE REVIEW

File: types/got/got-tests.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ got('todomvc.com')
 got('todomvc.com').cancel();
 
 got('todomvc.com', {json: true}).then(response => obj = response.body);
+got('todomvc.com', {json: true}).then(response => str = response.body);
```
Comment (BendingBender) [inline types/got/got-tests.ts]: Would be better to use a type assertion:
```typescript
got('todomvc.com', {json: true}).then(response => {
  response.body; // $ExpectType any
});
```

**Review by BendingBender — CHANGES_REQUESTED** [review]

**Review by BendingBender — CHANGES_REQUESTED** [review]
Please fix commented test.

File: types/got/got-tests.ts (hunk 1 above)
Comment (LinusU) [inline types/got/got-tests.ts]: Awesome! Did not know about that :)

**Comment by LinusU** [comment]
@BendingBender fixed 👌

**Comment by BendingBender** [comment]
:+1: Perfect!

**Pull request merged**